util/matchers.go

```diff
@@ -10,7 +10,8 @@ import (
 // SplitFiltersAndMatchers splits empty matchers off, which are treated as filters, see #220
 func SplitFiltersAndMatchers(allMatchers []*metric.LabelMatcher) (filters, matchers []*metric.LabelMatcher) {
 	for _, matcher := range allMatchers {
-		if matcher.Match("") {
+		// Test for empty filters. Other metrics such as NotEqual are not filters.
```

Comment:
I've gone through this again and understand why we are filtering out all matchers matching "". I will revert the change and describe in a comment why.